scipy/stats/tests/test_stats.py

```diff
@@ -3453,6 +3464,8 @@ def test_nan_policy(self):
         assert_raises(ValueError, stats.moment, x, nan_policy='raise')
         assert_raises(ValueError, stats.moment, x, nan_policy='foobar')
 
+    @skip_xp_backends('cupy', reasons=['cupy/cupy#8391'])
+    @pytest.mark.usefixtures("skip_xp_backends")
```

Comment:
I'm slightly fuzzy on why we sometimes need the `@pytest.mark.usefixtures("skip_xp_backends")` line and sometimes don't. It seems undesirable. It's probably been discussed before and I forgot or missed it - could you enlighten me?

Reply:
If every test in the file has `xp` as a parameter, it can go in `pytestmark` and thus only needs to occur once in the whole file.

If some tests do not have `xp` as a parameter (such as this ridiculously large `test_stats` file), `pytest` *can (although I'm not 100% sure always)* complain during collection. So it needs to be applied at a lower level.

Reply:
Ah okay, so it's some gap when a custom decorator is being applied to a custom mark. Yeah, `pytest` works in mysterious ways:)